go/def.bzl

```diff
@@ -389,9 +390,25 @@ def emit_go_link_action(ctx, importmap, transitive_libs, cgo_deps, lib,
   # Avoided -s on OSX but but it requires dsymutil to be on $PATH.
   # TODO(yugui) Remove this workaround once rules_go stops supporting XCode 7.2
   # or earlier.
-  cmds += ["export PATH=$PATH:/usr/bin"]
   cmds += [
+    "export PATH=$PATH:/usr/bin",
     "export GOROOT=$(pwd)/" + ctx.file.go_tool.dirname + "/..",
+    "STAMP_XDEFS=()",
+  ]
+
+  stamp_input = []
+  if ctx.attr.stamp and ctx.attr.linkstamp:
```

Comment:
should we maybe `fail` if `stamp` is true but `linkstamp` isn't set? otherwise it might be confusing why `stamp` isn't working.

(alternately, do we need both? or just use nonempty `linkstamp` as the setting?)